
From: Antonio Radici <antonio@debian.org>
Date: Thu, 27 Feb 2014 17:11:56 +0100
Subject: 228671-pipe-mime

Don't draw imap fetch progress if we aren't in visual mode.
Drawing progress leaves terminal in confusing state when
piping a message from pager to less(1).
See http://bugs.mutt.org/1771

Updated in Debian bug #569279

To apply to neomutt by:
From 719e02e7ecfd78f45a5543d21405f7b5a183184d Mon Sep 17 00:00:00 2001
From: Richard Russon <rich@flatcap.org>
Date: Fri, 6 May 2016 00:24:03 +0100
Subject: [PATCH 17/22] upstream/228671-pipe-mime

---
 imap/message.c | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

Index: neomutt-20160818/imap/message.c
===================================================================
--- neomutt-20160818.orig/imap/message.c
+++ neomutt-20160818/imap/message.c
@@ -401,7 +401,7 @@ int imap_fetch_message (struct Context *ctx, ME
   char path[_POSIX_PATH_MAX];
   char *pc;
   long bytes;
-  struct Progress progressbar;
+  struct Progress progressbar, *pbar;
   int uid;
   int cacheno;
   struct ImapCache *cache;
@@ -499,9 +499,15 @@ int imap_fetch_message (struct Context *ctx, ME
 	    imap_error ("imap_fetch_message()", buf);
 	    goto bail;
 	  }
-	  mutt_progress_init (&progressbar, _("Fetching message..."),
-			      MUTT_PROGRESS_SIZE, NetInc, bytes);
-	  if (imap_read_literal (msg->fp, idata, bytes, &progressbar) < 0)
+	  if (!isendwin())
+	  {
+	    mutt_progress_init (&progressbar, _("Fetching message..."),
+                               MUTT_PROGRESS_SIZE, NetInc, bytes);
+	    pbar = &progressbar;
+	  }
+	  else
+	    pbar = NULL;
+	  if (imap_read_literal (msg->fp, idata, bytes, pbar) < 0)
 	    goto bail;
 	  /* pick up trailing line */
 	  if ((rc = imap_cmd_step (idata)) != IMAP_CMD_CONTINUE)
